src/View/ViewContainer.vala

```diff
@@ -105,34 +138,51 @@ namespace Files.View {
 
         /* Initial location now set by Window.make_tab after connecting signals */
         public ViewContainer (View.Window win) {
-            window = win;
-            browser = new Browser ();
-
-            set_events (Gdk.EventMask.ENTER_NOTIFY_MASK | Gdk.EventMask.LEAVE_NOTIFY_MASK);
-            connect_signals ();
+            Object (
+                window: win
+            );
         }
 
-        ~ViewContainer () {
-            debug ("ViewContainer destruct");
-        }
-
-        private void connect_signals () {
+        construct {
+            browser = new Browser ();
             loading.connect ((loading) => {
                 is_loading = loading;
             });
 
-            button_press_event.connect (on_button_press_event);
-        }
-
-        private void connect_window_signals () {
-            if (window != null) {
-                window.folder_deleted.connect (on_folder_deleted);
-            }
+            var button_controller = new Gtk.GestureMultiPress (this) {
+                button = 0,
```

Comment:
Isn't this `0` by default?

Reply:
Yes, but I like to make it an explicit choice, at least at this stage.